app.rb

```diff
@@ -9,51 +9,41 @@
 KAFKA_TOPIC = ENV.fetch('KAFKA_TOPIC', 'messages')
 GROUP_ID = ENV.fetch('KAFKA_CONSUMER_GROUP', 'heroku-kafka-demo')
 
-def with_prefix(name)
-  "#{ENV['KAFKA_PREFIX']}#{name}"
-end
-
 def initialize_kafka
   tmp_ca_file = Tempfile.new('ca_certs')
   tmp_ca_file.write(ENV.fetch('KAFKA_TRUSTED_CERT'))
   tmp_ca_file.close
 
-  # This demo app connects to kafka on multiple threads.
-  # Right now ruby-kafka isn't thread safe, so we establish a new client
-  # for the producer and a different one for the consumer.
-  #
-  producer_kafka = Kafka.new(
-    seed_brokers: ENV.fetch('KAFKA_URL'),
-    ssl_ca_cert_file_path: tmp_ca_file.path,
-    ssl_client_cert: ENV.fetch('KAFKA_CLIENT_CERT'),
-    ssl_client_cert_key: ENV.fetch('KAFKA_CLIENT_CERT_KEY'),
-    ssl_verify_hostname: false,
-  )
-
-  $producer = producer_kafka.async_producer(delivery_interval: 1)
-
-  consumer_kafka = Kafka.new(
-    seed_brokers: ENV.fetch('KAFKA_URL'),
-    ssl_ca_cert_file_path: tmp_ca_file.path,
-    ssl_client_cert: ENV.fetch('KAFKA_CLIENT_CERT'),
-    ssl_client_cert_key: ENV.fetch('KAFKA_CLIENT_CERT_KEY'),
-    ssl_verify_hostname: false,
-  )
+  $producer = Rdkafka::Config.new({
+    :"bootstrap.servers" => ENV.fetch('KAFKA_URL').gsub('kafka+ssl://', ''),
+    :"security.protocol" => "ssl",
+    :"ssl.ca.location" => tmp_ca_file.path,
+    :"ssl.key.pem" => ENV.fetch('KAFKA_CLIENT_CERT_KEY'),
+    :"ssl.certificate.pem" => ENV.fetch('KAFKA_CLIENT_CERT'),
+    :"enable.ssl.certificate.verification" => false,
+  }).producer
 
   # Connect a consumer. Consumers in Kafka have a "group" id, which
   # denotes how consumers balance work. Each group coordinates
   # which partitions to process between its nodes.
   # For the demo app, there's only one group, but a production app
   # could use separate groups for e.g. processing events and archiving
   # raw events to S3 for longer term storage
-  #
-  $consumer = consumer_kafka.consumer(group_id: with_prefix(GROUP_ID))
+  $consumer = Rdkafka::Config.new({
+    :"bootstrap.servers" => ENV.fetch('KAFKA_URL').gsub('kafka+ssl://', ''),
+    :"security.protocol" => "ssl",
+    :"ssl.ca.location" => tmp_ca_file.path,
+    :"ssl.key.pem" => ENV.fetch('KAFKA_CLIENT_CERT_KEY'),
+    :"ssl.certificate.pem" => ENV.fetch('KAFKA_CLIENT_CERT'),
+    :"enable.ssl.certificate.verification" => false,
+    :"group.id" => GROUP_ID,
+  }).consumer
+
   $recent_messages = []
   start_consumer
-  start_metrics
 
   at_exit do
-    $producer.shutdown
+    $producer.close
```

Comment:
`$producer.flush` should be called before `$producer.close` to ensure all messages are sent to Kafka before closing the producer.

Reply:
Thanks. Updated.